web/admin.py

```
@@ -812,3 +821,141 @@ class PointsAdmin(admin.ModelAdmin):
+        ("Timestamps", {"fields": ("joined_at",), "classes": ("collapse",)}),
+    )
+    actions = ["approve_selected", "reject_selected"]
+    
+    def approve_selected(self, request: HttpRequest, queryset: QuerySet["ChapterMembership"]) -> None:
+        pending = queryset.filter(status='pending')
+        for membership in pending:
+            membership.approve(request.user)
+        self.message_user(request, f"{pending.count()} memberships approved.")
+    approve_selected.short_description = "Approve selected memberships"
+    
+    def reject_selected(self, request: HttpRequest, queryset: QuerySet["ChapterMembership"]) -> None:
```

Ruff (0.8.2)

Undefined name `HttpRequest`

(F821)